rs/tests/driver/src/util.rs

```diff
@@ -734,6 +734,26 @@ pub async fn assert_create_agent(url: &str) -> Agent {
         .unwrap_or_else(|err| panic!("Failed to create agent for {}: {:?}", url, err))
 }
 
+/// Initializes an `Agent` using the provided URL.
+/// The root key is fetched as part of the initialization in order
+/// to validate certificates from the replica.
+///
+/// The created agent will route call requests to the
+/// asynchronous V2 call endpoint.
+pub async fn assert_create_agent_using_call_v2(url: &str) -> Agent {
+    let start = Instant::now();
+    while start.elapsed() < READY_WAIT_TIMEOUT {
+        if let Ok(v) = create_agent(url).await {
+            return v;
```

Comment:
I don't follow this code. Won't this return an Agent that [uses the v3 endpoint](https://github.com/dfinity/ic/blob/a1dfb3e6977bdd59451d90b785707ff2fc618cab/rs/tests/driver/src/util.rs#L796)?

Reply:
Yes good catch. Forgot to change this line when copying the function below.

Reply:
Could you also briefly document this retrying business? Because I couldn't understand why you're not calling `agent_using_call_v2_endpoint()` directly.

Reply:
Yes, I can add a comment.

Just a note. The implementation is just copy pasted from the function below, [assert_create_agent_with_identity](assert_create_agent_with_identity.

Reply:
I am not sure why the retrying logic is there either in the original function.

The reason I created this method with the retrying logic is that the [create_canister_agent](https://github.com/dfinity/ic/blob/a1dfb3e6977bdd59451d90b785707ff2fc618cab/rs/tests/driver/src/canister_agent.rs#L39) method that was originally used, calls [assert_create_agent_with_identity](https://github.com/dfinity/ic/pull/737/files/db870a50c544c87e3a8897c7e592d1dd44ff7f97#diff-4a90e804c450aa8c0f067d3f8b2b9f4b950d9913fa363224612970b4c861ad1bR758-R773) which has the retrying logic.

Either way, this is just a temporary function, and I will look into enabling